x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/form/config.tsx

```diff
@@ -0,0 +1,819 @@
+      stepKey="step2"
+    />
+  ),
+};
+const MONITOR_SCRIPT_STEP: Step = {
+  title: 'Add a script',
+  children: (
+    <StepFields
+      description="Use Elastic Script Recorder to generate a script and then upload it. Alternatively, you can write your own Playwright script and paste it in the script editor."
+      stepKey="step3"
+    />
+  ),
+};
+
+export const ADD_MONITOR_STEPS: StepMap = {
+  [FormMonitorType.MULTISTEP]: [MONITOR_TYPE_STEP, MONITOR_DETAILS_STEP, MONITOR_SCRIPT_STEP],
+  [FormMonitorType.SINGLE]: [MONITOR_TYPE_STEP, MONITOR_DETAILS_STEP],
+  [FormMonitorType.HTTP]: [MONITOR_TYPE_STEP, MONITOR_DETAILS_STEP],
+  [FormMonitorType.ICMP]: [MONITOR_TYPE_STEP, MONITOR_DETAILS_STEP],
+  [FormMonitorType.TCP]: [MONITOR_TYPE_STEP, MONITOR_DETAILS_STEP],
+};
+
+export const EDIT_MONITOR_STEPS: StepMap = {
+  [FormMonitorType.MULTISTEP]: [MONITOR_SCRIPT_STEP, MONITOR_DETAILS_STEP, MONITOR_SCRIPT_STEP],
+  [FormMonitorType.SINGLE]: [MONITOR_DETAILS_STEP],
+  [FormMonitorType.HTTP]: [MONITOR_DETAILS_STEP],
+  [FormMonitorType.ICMP]: [MONITOR_DETAILS_STEP],
+  [FormMonitorType.TCP]: [MONITOR_DETAILS_STEP],
+};
+
+export const MONITOR_TYPE_CONFIG = {
+  [FormMonitorType.MULTISTEP]: {
+    id: 'syntheticsMonitorTypeMultiStep',
+    label: 'Multistep',
+    value: FormMonitorType.MULTISTEP,
+    descriptionTitle: 'Multistep Browser Journey',
+    description:
+      'A lightweight API check to validate the availability of a web service or endpoint.',
+    link: '#',
+    icon: 'videoPlayer',
+    beta: true,
+  },
+  [FormMonitorType.SINGLE]: {
+    id: 'syntheticsMonitorTypeSingle',
+    label: 'Single Page',
+    value: FormMonitorType.SINGLE,
+    descriptionTitle: 'Single Page Browser Test',
+    description:
+      'A lightweight API check to validate the availability of a web service or endpoint.',
+    link: '#',
+    icon: 'videoPlayer',
+    beta: true,
+  },
+  [FormMonitorType.HTTP]: {
+    id: 'syntheticsMonitorTypeHTTP',
+    label: 'HTTP Ping',
+    value: FormMonitorType.HTTP,
+    descriptionTitle: 'HTTP Ping',
+    description:
+      'A lightweight API check to validate the availability of a web service or endpoint.',
+    link: '#',
+    icon: 'online',
+    beta: false,
+  },
+  [FormMonitorType.ICMP]: {
+    id: 'syntheticsMonitorTypeICMP',
+    label: 'ICMP Ping',
+    value: FormMonitorType.ICMP,
+    descriptionTitle: 'ICMP Ping',
+    description:
+      'A lightweight API check to validate the availability of a web service or endpoint.',
+    link: '#',
+    icon: 'online',
+    beta: false,
+  },
+  [FormMonitorType.TCP]: {
+    id: 'syntheticsMonitorTypeTCP',
+    label: 'TCP Ping',
+    value: FormMonitorType.TCP,
+    descriptionTitle: 'TCP Ping',
+    description:
+      'A lightweight API check to validate the availability of a web service or endpoint.',
+    link: '#',
+    icon: 'online',
+    beta: false,
+  },
+};
+
+export const FIELD: Record<string, FieldMeta> = {
```

Comment:
Configuration for each individual field, mapping them back to each key and defining the rendered component as well as additional props, validation, error messages, etc.